tests/scripts/all.sh

```diff
@@ -817,6 +823,16 @@ pre_generate_files() {
     fi
 }
 
+check_test_extended () {
+    # Usage: put "check_test_extended || return" at the top of every test that should
+    # only run when extended tests are enabled. Doing this instead of using a
+    # supports_xxx function stops the CI from treating this as a not-tested component.
+    if [ -z ${MBEDTLS_TEST_EXTENDED+x} ]; then
+        msg "Test is a no-op because MBEDTLS_TEST_EXTENDED is not set"
```

Comment:
and let's say "Test SKIPPED for abridged test run" (or similar, but please don't call it a no-op, and using the shell variable name adds a layer of indirection to the message